dnainator-javafx/src/main/java/nl/tudelft/dnainator/javafx/drawables/SemanticDrawable.java

```diff
@@ -1,76 +1,69 @@
 package nl.tudelft.dnainator.javafx.drawables;
 
+import javafx.beans.property.IntegerProperty;
+import javafx.beans.property.SimpleIntegerProperty;
 import javafx.geometry.Bounds;
 import javafx.scene.Group;
+import nl.tudelft.dnainator.annotation.Range;
 
 /**
  * An abstract drawable that enables a semantic display of data.
  */
 public abstract class SemanticDrawable extends Group {
+	protected static final double RANK_WIDTH = 100;
+	protected IntegerProperty minRank = new SimpleIntegerProperty(0, "minRank");
+	protected IntegerProperty maxRank = new SimpleIntegerProperty(0, "maxRank");
 	protected Group content;
-	protected Group childContent;
 
 	/**
 	 * Creates a new {@link SemanticDrawable}.
-	 * @param content Top-level content.
-	 * @param childContent Lower-level content.
+	 * @param content The content to display.
 	 */
-	public SemanticDrawable(Group content, Group childContent) {
+	public SemanticDrawable(Group content) {
 		this.content = content;
-		this.childContent = childContent;
-		getChildren().addAll(content, childContent);
+		getChildren().addAll(content);
 	}
 
 	/**
-	 * Load the {@link Drawable} content of the view itself. The overriding class must make sure
+	 * Load the {@link Drawable} content of the view. The overriding class must make sure
 	 * that this method gets called either in the constructor or right after!
-	 * @param bounds The bounds of the viewport.
+	 * @param range The minimum and maximum range that are in view at the moment.
+	 * @param zoom The current (geometric) zoom level.
 	 */
-	protected abstract void loadContent(Bounds bounds);
+	protected abstract void loadContent(Range range, double zoom);
 
 	/**
-	 * Load the drawable content of the view's children, given the bounds of the viewport.
-	 * @param bounds The bounds of the viewport.
+	 * Update method that should be called after scaling.
+	 * This method checks how zoomed in we are by transforming bounds to root coordinates,
+	 * and then dynamically adds and deletes items in the JavaFX scene graph.
+	 * @param bounds	The bounds of the new viewport.
+	 * @param zoom      The current (geometric) zoom level.
 	 */
-	protected abstract void loadChildren(Bounds bounds);
+	public void update(Bounds bounds, double zoom) {
+		minRank.set((int) (Math.max(bounds.getMinX() / RANK_WIDTH, 0)));
+		maxRank.set((int) (RANK_WIDTH + bounds.getMaxX() / RANK_WIDTH));
+		loadContent(new Range(minRank.get(), maxRank.get()), zoom);
```

Comment:
In this class you use properties, whereas you use a similar implementation without properties in the subclass. Then you only set them in `needRedraw`, why?